.github/workflows/ci.yaml

```diff
@@ -6,13 +6,20 @@ on:
   release:
     types:
       - released
-  pull_request:
+  pull_request_target:
     branches:
       - master
+    types: 
+      - labeled
 jobs:
+  check-is-safe-to-run:
+    name: Check the pull request checks are safe to be run
+    if: contains(github.event.pull_request.labels.*.name, 'Safe to run')
```

Comment:
This job runs only if the label is added and only runs in that case.

Then we make each other job run only if this job ran too